iam.tf

```diff
@@ -51,9 +51,9 @@ resource "aws_iam_role_policy_attachment" "existing_policies_for_eks_workers_rol
 # Create a CNI policy that is a merger of AmazonEKS_CNI_Policy and required IPv6 permissions
 # https://github.com/SummitRoute/aws_managed_policies/blob/master/policies/AmazonEKS_CNI_Policy
 # https://docs.aws.amazon.com/eks/latest/userguide/cni-iam-role.html#cni-iam-role-create-ipv6-policy
-
+# https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html
 data "aws_iam_policy_document" "ipv6_eks_cni_policy" {
-  count = local.create_role && var.node_role_cni_policy_enabled ? 1 : 0
+  count = local.create_role && var.node_role_cni_policy_enabled || local.create_role && can(regex("WINDOWS", var.ami_type)) ? 1 : 0
```

Comment:
I'm checking this one.
To run windows you have to run linux nodes as well to provide the funtions "currently" not supported on windows by k8s (2023 is supposed to be a 'focus on windows' year so who knows)
The changes were based on our working cluster. 
The minimum requirements are the policy on the cluster (hence my pr [on that](https://github.com/cloudposse/terraform-aws-eks-cluster/pull/175), and the policy and taints on windows nodes and a config map for IPAM.
The policy is to allow assignment of IPs and I _think_ I set it to use it on windows so that would happen.
I need to retest as we dont use the built in role and I havent looked at this since January 😉